### PR TITLE
Fix #674: Use dynamic retrieve of search field in meta data editor

### LINKED ISSUE #674: Meta-data editor not fetching serach fields from catalogue plug-in
Search fields are still hard-coded [in `Strukturdaten.jsp`](https://github.com/kitodo/kitodo-production/blob/master/Kitodo/src/main/webapp/pages/incMeta/Strukturdaten.jsp#L175-L179). Affected function is _Add docstructs from Opac_ on the _Structure data_ page of the meta-data editor.

_BTW, does someone know how this function is used and what it does?_

unbenannt

### PR DESCRIPTION
**Pull request by henning-gerhardt**
Remove hard coded values for catalogue based search from meta data editor.

### CODE REVIEW

File: Goobi/pages/incMeta/Strukturdaten.jsp [hunk 1]
```diff
@@ -204,16 +204,12 @@
 
                                 <%-- uses goobi_opac.xml --%>
 
-                                <h:selectOneMenu id="katalogauswahl" value="#{Metadaten.opacKatalog}" style="display:inline; margin-left:7px">
+                                <h:selectOneMenu id="katalogauswahl" value="#{Metadaten.opacKatalog}" style="display:inline; margin-left:7px" onchange="submit();">
```
Comment (matthias-ronge) [inline Goobi/pages/incMeta/Strukturdaten.jsp]: Please use ajax to reload the form: Remove `onchange="submit();"`, add `<a4j:support event="onchange" reRender="feldauswahl" />` between `<si:selectItems … />` and `</h:selectOneMenu>`.

File: Goobi/src/de/sub/goobi/metadaten/Metadaten.java [hunk 2]
```diff
@@ -3381,4 +3382,14 @@ public void setAddMetaDataValue(String addMetaDataValue) {
...
         return result;
     }
+
+    /**
+     * @see @link{de.sub.goobi.forms.ProzesskopieForm.getSearchFields}
+     *
+     * @return A map containing the search fields for the currently selected OPAC
+     */
+    public HashMap<String, String> getSearchFields() {
+        return ProzesskopieForm.getSearchFieldsForCatalogue(opacKatalog);
```
Comment (matthias-ronge) [inline Goobi/src/de/sub/goobi/metadaten/Metadaten.java]: Returns an empty list if opacKatalog is still `null` (first start of meta-data editor). Fix:

    String catalogue = opacKatalog;
    if(catalogue == null) {
        List<String> allOpacCatalogues = ProzesskopieForm.getAllOpacCatalogues();
        if(!allOpacCatalogues.isEmpty()){
            catalogue = allOpacCatalogues.get(0);
        }
    }
    return ProzesskopieForm.getSearchFieldsForCatalogue(catalogue);

`ProzesskopieForm.getAllOpacCatalogues()` can and needs to be declared `static` to do so.

**Comment by matthias-ronge** [comment]
Looks ok now :)

**Comment by henning-gerhardt** [comment]
@matthias-ronge : I integrated your suggestions and improvements. 

`ProzesskopieForm.getAllOpacCatalogues`  could not be static as this method is overwritten in `CopyProcess` and throws an exception (`javax.faces.el.PropertyNotFoundException: Bean: de.sub.goobi.forms.ProzesskopieForm, property: allOpacCatalogues` ... full stacktrace is available) in meta data editor.

**Pull request merged**